packages/hatchify-core/src/assembler.ts

```diff
@@ -0,0 +1,115 @@
+import { integer } from "./dataTypes"
+import type {
+  FinalAttribute,
+  FinalNumberORM,
+  PartialAttribute,
+  PartialNumberControlType,
+  PartialNumberORM,
+} from "./types"
+
+export interface PartialSchema {
+  name: string
+  id?: PartialAttribute<
+    PartialNumberORM,
+    PartialNumberControlType,
+    number,
+    FinalNumberORM
+  >
+  attributes: {
+    [attributeName: string]: PartialAttribute<
+      PartialNumberORM,
+      PartialNumberControlType,
+      number,
+      FinalNumberORM
+    >
+  }
+}
+export interface FinalSchema {
+  name: string
+  id: FinalAttribute<
+    PartialNumberORM,
+    PartialNumberControlType,
+    number,
+    FinalNumberORM
+  >
+  attributes: {
+    [attributeName: string]: FinalAttribute<
+      PartialNumberORM,
+      PartialNumberControlType,
+      number,
+      FinalNumberORM
+    >
+  }
+}
+
+type PartialSchemaWithPrimaryAttribute = Omit<PartialSchema, "id"> & {
+  id: PartialAttribute<
+    PartialNumberORM,
+    PartialNumberControlType,
+    number,
+    FinalNumberORM
+  >
+}
+
+export function assembler(schemas: { [schemaName: string]: PartialSchema }): {
+  [schemaName: string]: FinalSchema
+} {
+  return Object.entries(schemas).reduce(
+    (acc, [schemaName, schema]) => ({
+      ...acc,
+      [schemaName]: finalizeSchema(setDefaultPrimaryAttribute(schema)),
+    }),
+    {},
+  )
+}
+
+function finalizeSchema(
+  schema: PartialSchemaWithPrimaryAttribute,
+): FinalSchema {
+  return {
+    ...schema,
+    id: schema.id.finalize(),
+    attributes: Object.entries(schema.attributes).reduce(
+      (acc, [attributeName, attribute]) => ({
+        ...acc,
+        [attributeName]: attribute.finalize(),
+      }),
+      {},
+    ),
+  }
+}
+
+function setDefaultPrimaryAttribute(
+  schema: PartialSchema,
+): PartialSchemaWithPrimaryAttribute {
+  return {
```

Comment:
What if we "forget" to add some defaults to `getDefaultPrimaryAttribute()`?

We could avoid that with something like:

```js
const id = schema.id || integer();

return {
  ...schema,
  id: {
    ...id,
          control: {
            ...id.control,
            primary: true,
            allowNull: false,
          },
          orm: {
            ...id.orm,
            sequelize: {
              ...id.orm.sequelize,
              primaryKey: true,
              allowNull: false,
            },
          },
  }
}
```